File: tests/nodejs/cases/case_015_publish_ticket.js
/**
 * Task case 015: publish_ticket
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case015PublishTicket(input) {
  // Task: publish ticket records and prepare transport-ready payload.
  const taskName = 'publish_ticket';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'ticket',
    stage: 'publish',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  // SAFE: payload remains local; only primitive checksum string is returned.
  return payload.checksum;
}

module.exports = {
  case015PublishTicket,
};

File: tests/python/cases/case_015_publish_ticket.py
"""
Task case 015: publish ticket payload for downstream business processing.
"""

RETAINED_CACHE = {}
RETAINED_AUDIT = []
RETAINED_HANDLERS = []

def case_015_publish_ticket(input_data):
    # Task: publish ticket records and prepare transport-ready payload.
    task_name = 'publish_ticket'
    raw = input_data or 'sample'
    payload = {
        'task': task_name,
        'entity': 'ticket',
        'stage': 'publish',
        'input': raw,
        'checksum': f'{task_name}:{len(raw)}',
    }
    # SAFE: payload remains local; only primitive checksum string is returned.
    return payload['checksum']
